moveit_core/robot_state/src/robot_state.cpp

```diff
@@ -998,41 +1000,78 @@ const Eigen::Isometry3d& RobotState::getFrameTransform(const std::string& frame_
   BOOST_VERIFY(checkLinkTransforms());
 
   static const Eigen::Isometry3d IDENTITY_TRANSFORM = Eigen::Isometry3d::Identity();
+  // Check if frame is in robot links
   if (frame_id == robot_model_->getModelFrame())
     return IDENTITY_TRANSFORM;
   if (robot_model_->hasLinkModel(frame_id))
   {
     const LinkModel* lm = robot_model_->getLinkModel(frame_id);
     return global_link_transforms_[lm->getLinkIndex()];
   }
+
+  // Check names of the AttachedBody objects themselves
   std::map<std::string, AttachedBody*>::const_iterator jt = attached_body_map_.find(frame_id);
   if (jt == attached_body_map_.end())
   {
     ROS_ERROR_NAMED(LOGNAME, "Transform from frame '%s' to frame '%s' is not known "
                              "('%s' should be a link name or an attached body's id).",
+    ROS_DEBUG_NAMED(LOGNAME,
+                    "Transform from frame '%s' to frame '%s' is not known to the robot state "
+                    "('%s' should be a link name, an attached body, or the name of an attached body's subframe).",
                     frame_id.c_str(), robot_model_->getModelFrame().c_str(), frame_id.c_str());
-    return IDENTITY_TRANSFORM;
   }
+
+  // If frame_id contains a separating slash, look through subframes of the AttachedBody objects
+  if (frame_id.find("/")!=std::string::npos)
```

Comment:
Here, the order of searching for attached bodies and subframes is reversed!

Reply:
I put it in front because checking for only the separator in the name is a pretty light operation. I understood this approach to be the result of the discussion further up.